packages/server/src/commands/build.ts

```diff
@@ -1,4 +1,104 @@
-export default function build() {
-  console.error("not implemented yet")
-  process.exit(1)
+import path from "path"
+import fs from "fs"
+import { ServerCommandBuild } from "@samen/dev"
+import ts, { CompilerOptions } from "typescript"
+import {
+  generateAppDeclarationFile,
+  generateRPCProxy,
+  generateProductionServer,
+  MissingSamenFileError,
+  MissingTSConfigFile,
+  parseSamenApp,
+} from "@samen/core"
+
+export default function build(command: ServerCommandBuild) {
+  const projectPath = process.cwd()
+
+  const tsConfigFilePath = ts.findConfigFile(
+    projectPath,
+    ts.sys.fileExists,
+    "tsconfig.json",
+  )
+
+  if (!tsConfigFilePath) {
+    throw new MissingTSConfigFile(projectPath)
+  }
+
+  const tsConfig: ts.ParsedCommandLine | undefined =
+    ts.getParsedCommandLineOfConfigFile(
+      tsConfigFilePath,
+      undefined,
+      ts.sys as any,
+    )
+
+  if (!tsConfig) {
+    throw new Error("Can't parse tsconfig.json")
+  }
+
+  const hasES2015 = tsConfig.options.lib?.some((lib: string) =>
+    /es2015/.test(lib.toLowerCase()),
```

Comment:
I think you intend to also match for the individual libraries like `es2015.promise` right?

Reply:
fixed!